setup.py

```diff
@@ -309,10 +317,49 @@ def find_eigen_dir(output=False):
                 print("\n  ", dir, "  (yes)")
             return dir
         if output: print("  (no)")
+
     if output:
-        print("Could not find Eigen.  Make sure it is installed either in a standard ")
-        print("location such as /usr/local/include, or the installation directory is either in ")
-        print("your C_INCLUDE_PATH or EIGEN_DIR environment variable.")
+        print("Could not find Eigen in any of the standard locations.")
+        print("Will now try to download it from bitbucket.org. This requires an internet")
+        print("connection, so it will fail if you are currently offline.")
+        print("If Eigen is installed in a non-standard location, and you want to use that")
+        print("instead, you should make sure the right directory is either in your")
+        print("C_INCLUDE_PATH or specified in an EIGEN_DIR environment variable.")
+
+    try:
+        dir = 'downloaded_eigen'
+        if os.path.isdir(dir):
+            # If this exists, it was tried above and failed.  Something must be wrong with it.
```

Comment:
I think a user message like "Previous attempt at downloading Eigen found.  Deleting and starting again." would be appropriate here.

Reply:
Yes.  Good idea.  Done.